openhands/controller/agent.py

```diff
@@ -38,6 +40,34 @@ def __init__(
         self._complete = False
         self.prompt_manager: 'PromptManager' | None = None
         self.mcp_tools: list[dict] = []
+        self.tools: list = []
+
+    def get_initial_message(self) -> 'SystemMessageAction':
+        """
+        Returns a SystemMessageAction containing the system message and tools.
+        This will be added to the event stream as the first message.
+
+        Returns:
+            SystemMessageAction: The system message action with content and tools
+        """
+        # Import here to avoid circular imports
+        from openhands.events.action.message import SystemMessageAction
+
+        if not self.prompt_manager:
+            raise ValueError(
+                'Prompt manager must be initialized before getting initial message'
+            )
+
+        system_message = self.prompt_manager.get_system_message()
+
+        # Get tools if available
+        tools = getattr(self, 'tools', None)
+
+        system_message_action = SystemMessageAction(content=system_message, tools=tools)
+        # Set the source attribute
+        system_message_action._source = EventSource.AGENT  # type: ignore
+
+        return system_message_action
```

Comment:
We should make sure all the agent in `openhands/agenthub` uses this SystemMessageAction to get system prompt when applicable.

Reply:
From OpenHands:

> Verified that all agents in openhands/agenthub are compatible with the changes:

> Found that CodeActAgent._get_messages already handles SystemMessageAction correctly
> No other agents override get_initial_message or use process_initial_messages